src/EspMQTTClient.cpp

```diff
@@ -663,42 +663,57 @@ void EspMQTTClient::processDelayedExecutionRequests()
 }
 
 /**
- * Matching MQTT topics, handling the eventual presence of a single wildcard character
+ * Matching MQTT topics, handling the eventual presence of wildcards character
+ * It doesn't validate the correctness of the topic pattern.
  *
- * @param topic1 is the topic may contain a wildcard
+ * @param topic1 is the topic may contain wildcard(s)
  * @param topic2 must not contain wildcards
  * @return true on MQTT topic match, false otherwise
  */
-bool EspMQTTClient::mqttTopicMatch(const String &topic1, const String &topic2) 
+bool EspMQTTClient::mqttTopicMatch(const String &topic1, const String &topic2)
 {
-  int i = 0;
+  const char *topic1_p = topic1.begin();
+  const char *topic1_end = topic1.end();
+  const char *topic2_p = topic2.begin();
+  const char *topic2_end = topic2.end();
 
-  if((i = topic1.indexOf('#')) >= 0) 
-  {
-    String t1a = topic1.substring(0, i);
-    String t1b = topic1.substring(i+1);
-    if((t1a.length() == 0 || topic2.startsWith(t1a)) &&
-       (t1b.length() == 0 || topic2.endsWith(t1b)))
+  while (topic1_p < topic1_end && topic2_p < topic2_end) {
+    if (*topic1_p == '#'){
```

Comment:
Align with the rest of the formatting
```suggestion
  while (topic1_p < topic1_end && topic2_p < topic2_end)
  {
    if (*topic1_p == '#')
    {
```